diracx/values.yaml

```diff
@@ -83,6 +84,53 @@ diracx:
   # Which DiracX OpenSearch DBs are used?
   osDatabases:
     - JobParametersDB
+  ingress:
+    enabled: true
+    className: "nginx"
+    annotations:
+      # kubernetes.io/ingress.class: nginx
+      # kubernetes.io/tls-acme: "true"
+      # Needed if you want to use a base path different from /
+      # #Use rewrite rules https://kubernetes.github.io/ingress-nginx/examples/rewrite/
+      nginx.ingress.kubernetes.io/use-regex: "true"
+      nginx.ingress.kubernetes.io/rewrite-target: /$2
```

Comment:
This is stripping the /api prefix right? If so this binds us strongly to nginx as most other load balancers (e.g haproxy in openshift) don’t support this.

Reply:
> This is stripping the /api prefix right? 

Yes. I kind of followed https://fastapi.tiangolo.com/advanced/behind-a-proxy/

> If so this binds us strongly to nginx as most other load balancers (e.g haproxy in openshift) don’t support this.

Really? I have no knowledge about any other load balancers, but it does not seem to be a niche use case, does it?
I just found this example for instance: https://discourse.haproxy.org/t/need-to-remove-1st-part-of-url/3990/2

Do you have any other solution in mind?

Reply:
I might be wrong but I thought I had seen it in the openshift docs. The alternative would be to route based on the path but don’t rewrite it (I.e fastapi is aware that it’s serving under /api/)

Reply:
Looks like it might be supported now since the move to OpenShift 4 (which CERN did last year): https://mattermost.web.cern.ch/it-dep/pl/efkd4preufbdxy55cgt7876cfa

> So, yesterday I spent some time trying what you are saying, namely redirecting requests for diracx through /api without rewriting the path. Here is the configuration I used:

I was thinking more like modifying this line: https://github.com/DIRACGrid/diracx/blob/690723bf1fec6fee7af66edcfd9973f63d46b230/src/diracx/routers/__init__.py#L137

```
         app.include_router(
             router,
-            prefix=f"/{system_name}",
+            prefix=f"/api/{system_name}",
             tags=[system_name],
             dependencies=dependencies,
         )
```

Reply:
We can then exclude `/.well-known/` from the prefix (which should be at the root of the domain as specified at the end of [RFC-8615 Section 3](https://www.rfc-editor.org/rfc/rfc8615.html#section-3)).

Reply:
Indeed, this could work too!

Do you have any preference? I don't really have one.
Stripping requests from the `ingress nginx` and letting `diracx` as it is now adds flexibility but also complexity.
Modifying the `diracx` prefix of the routes is probably less flexible, but we can have a simpler `ingress`.

Reply:
I lean towards making it hardcoded in fastapi as:

* I fear people will have trouble configuring the ingress so the more we can avoid installation specific options the better.
* I find applications rarely support rewriting prefixes correctly (see the issue with `/openapi.json`).
* Users will then set the `DIRACX_URL` variable to the same value as the webapp URL so it's easier than discovering they need to add a `/api/` suffix. (This one we can workaround with good error handling/messages, but if we can avoid it by design then it's even better.)

Reply:
Though I might be wrong in the opinion, so feel free to argue against me if you see things differently! 😆

Reply:
I see your point and I agree with you, I assume most people will just have to deal with `Values.yaml` to configure the existing Kubernetes components. Thus, the simplest values, the best.
+ I don't see any specific use case where someone would like to redirect requests differently (they could still modify their `ingress` accordingly).

I will change my PR then.